packages/performance/src/services/transport_service.ts

```diff
@@ -114,60 +99,13 @@ function dispatchQueueEvents(): void {
   };
   /* eslint-enable camelcase */
 
-  sendEventsToFl(data, staged).catch(() => {
-    // If the request fails for some reason, add the events that were attempted
-    // back to the primary queue to retry later.
-    queue = [...staged, ...queue];
-    remainingTries--;
-    consoleLogger.info(`Tries left: ${remainingTries}.`);
-    processQueue(DEFAULT_SEND_INTERVAL_MS);
-  });
+  postToFlEndpoint(data);
 }
 
-function sendEventsToFl(
-  data: TransportBatchLogFormat,
-  staged: BatchEvent[]
-): Promise<void> {
-  return postToFlEndpoint(data)
-    .then(res => {
-      if (!res.ok) {
-        consoleLogger.info('Call to Firebase backend failed.');
-      }
-      return res.json();
-    })
-    .then(res => {
-      // Find the next call wait time from the response.
-      const transportWait = Number(res.nextRequestWaitMillis);
-      let requestOffset = DEFAULT_SEND_INTERVAL_MS;
-      if (!isNaN(transportWait)) {
-        requestOffset = Math.max(transportWait, requestOffset);
-      }
-
-      // Delete request if response include RESPONSE_ACTION_UNKNOWN or DELETE_REQUEST action.
-      // Otherwise, retry request using normal scheduling if response include RETRY_REQUEST_LATER.
-      const logResponseDetails: LogResponseDetails[] = res.logResponseDetails;
-      if (
-        Array.isArray(logResponseDetails) &&
-        logResponseDetails.length > 0 &&
-        logResponseDetails[0].responseAction === 'RETRY_REQUEST_LATER'
-      ) {
-        queue = [...staged, ...queue];
-        consoleLogger.info(`Retry transport request later.`);
-      }
-
-      remainingTries = DEFAULT_REMAINING_TRIES;
-      // Schedule the next process.
-      processQueue(requestOffset);
-    });
-}
-
-function postToFlEndpoint(data: TransportBatchLogFormat): Promise<Response> {
+function postToFlEndpoint(data: TransportBatchLogFormat): boolean {
   const flTransportFullUrl =
     SettingsService.getInstance().getFlTransportFullUrl();
-  return fetch(flTransportFullUrl, {
-    method: 'POST',
-    body: JSON.stringify(data)
-  });
+  return navigator.sendBeacon(flTransportFullUrl, JSON.stringify(data));
```

Comment:
We should have a fall back to fetch if sendBeacon fails (eg. payload is > then the ~64KB limit)

```
(navigator.sendBeacon && navigator.sendBeacon(flTransportFullUrl, JSON.stringify(data))) ||
    fetch(flTransportFullUrl, {
        method: 'POST',
        body: JSON.stringify(data),
        keepalive: true
     }
     );
```

Reply:
Done